shell/platform/windows/win32_window.cc

```diff
@@ -4,9 +4,18 @@
 
 #include "flutter/shell/platform/windows/win32_window.h"
 
+#include "dpi_utils.h"
+
 namespace flutter {
 
-Win32Window::Win32Window() {}
+Win32Window::Win32Window() {
+  // Get the DPI of the primary monitor as the initial DPI. If Per-Monitor V2 is
+  // supported, |current_dpi_| should be updated in the
+  // kWmDpiChangedBeforeParent message.
+  const POINT target_point = {static_cast<LONG>(0), static_cast<LONG>(0)};
+  HMONITOR monitor = MonitorFromPoint(target_point, MONITOR_DEFAULTTOPRIMARY);
+  current_dpi_ = GetDpiForMonitor(monitor);
```

Comment:
You may as well use `GetDpiForWindow(nullptr)` here since it already has this logic, and getting the primary monitor for nullptr is a documented feature of that API.